layers/buffer_validation.cpp

```diff
@@ -5339,88 +5612,122 @@ bool CoreChecks::ValidateBufferBounds(const IMAGE_STATE *image_state, const BUFF
+    skip |= ValidateCmd(cb_node, cmd_type, func_name);
 
     // Command pool must support graphics, compute, or transfer operations
     const auto pPool = cb_node->command_pool.get();
 
     VkQueueFlags queue_flags = GetPhysicalDeviceState()->queue_family_properties[pPool->queueFamilyIndex].queueFlags;
 
     if (0 == (queue_flags & (VK_QUEUE_GRAPHICS_BIT | VK_QUEUE_COMPUTE_BIT | VK_QUEUE_TRANSFER_BIT))) {
-        skip |=
-            LogError(cb_node->createInfo.commandPool, "VUID-vkCmdCopyImageToBuffer-commandBuffer-cmdpool",
-                     "Cannot call vkCmdCopyImageToBuffer() on a command buffer allocated from a pool without graphics, compute, "
-                     "or transfer capabilities..");
-    }
-    skip |= ValidateImageBounds(src_image_state, regionCount, pRegions, "vkCmdCopyImageToBuffer()",
-                                "VUID-vkCmdCopyImageToBuffer-pRegions-00182");
-    skip |= ValidateBufferBounds(src_image_state, dst_buffer_state, regionCount, pRegions, "vkCmdCopyImageToBuffer()",
-                                 "VUID-vkCmdCopyImageToBuffer-pRegions-00183");
-
-    skip |= ValidateImageSampleCount(src_image_state, VK_SAMPLE_COUNT_1_BIT, "vkCmdCopyImageToBuffer(): srcImage",
-                                     "VUID-vkCmdCopyImageToBuffer-srcImage-00188");
-    skip |= ValidateMemoryIsBoundToImage(src_image_state, "vkCmdCopyImageToBuffer()", "VUID-vkCmdCopyImageToBuffer-srcImage-00187");
-    skip |=
-        ValidateMemoryIsBoundToBuffer(dst_buffer_state, "vkCmdCopyImageToBuffer()", "VUID-vkCmdCopyImageToBuffer-dstBuffer-00192");
+        vuid =
+            is_2khr ? "VUID-vkCmdCopyImageToBuffer2KHR-commandBuffer-cmdpool" : "VUID-vkCmdCopyImageToBuffer-commandBuffer-cmdpool";
+        skip |= LogError(cb_node->createInfo.commandPool, vuid,
+                         "Cannot call %s on a command buffer allocated from a pool without graphics, compute, "
+                         "or transfer capabilities.",
+                         func_name);
+    }
+    vuid = is_2khr ? "VUID-vkCmdCopyImageToBuffer2KHR-pRegions-00182" : "VUID-vkCmdCopyImageToBuffer-pRegions-00182";
+    skip |= ValidateImageBounds(src_image_state, regionCount, pRegions, func_name, vuid);
+    vuid = is_2khr ? "VUID-vkCmdCopyImageToBuffer2KHR-pRegions-00183" : "VUID-vkCmdCopyImageToBuffer-pRegions-00183";
+    skip |= ValidateBufferBounds(src_image_state, dst_buffer_state, regionCount, pRegions, func_name, vuid);
+
+    vuid = is_2khr ? "VUID-vkCmdCopyImageToBuffer2KHR-srcImage-00188" : "VUID-vkCmdCopyImageToBuffer-srcImage-00188";
+    const char *location = is_2khr ? "vkCmdCopyImageToBuffer2KHR(): srcImage" : "vkCmdCopyImageToBuffer(): srcImage";
+    skip |= ValidateImageSampleCount(src_image_state, VK_SAMPLE_COUNT_1_BIT, location, vuid);
+
+    vuid = is_2khr ? "VUID-vkCmdCopyImageToBuffer2KHR-srcImage-00187" : "VUID-vkCmdCopyImageToBuffer-srcImage-00187";
```

Comment:
Alot of these VUs seems to be a copy and paste error (hopefully not a spec common validity file error)

The copy2 is `VUID-struct-*` while copy1 is `VUID-command-*`

```suggestion
    vuid = is_2khr ? "VUID-VkCopyImageToBufferInfo2KHR-srcImage-00187" : "VUID-vkCmdCopyImageToBuffer-srcImage-00187";
```

Reply:
@mark-lunarg is `python3 scripts/vk_validation_stats.py build/Vulkan-Headers/registry/validusage.json -c` not being ran in CI? I thought that would be ran and catch these

Reply:
@sfricke-samsung, I don't believe it is anymore, as it was somehow affecting the header updates.  It IS run at header update time.

Reply:
that's fine, I guess more of a suggestion for @jackohound for this PR as it tackles a lot of new VUIDs and I know how unfun it is to start checking them by hand

Reply:
The tests should take care of some of that, but yeah, I understand.

Reply:
Thanks, have them all cleaned up, and will push an update, and yea the tool was helpful.  I see a few remaining, but unrelated to this PR.